proposals/list-patterns.md

```diff
@@ -0,0 +1,142 @@
+# List patterns
+
+## Summary
+
+Lets you to match an array or a list with a sequence of patterns e.g. `array is {1, 2, 3}` will match an integer array of the length three with 1, 2, 3 as its elements, respectively.
+
+## Detailed Design
+
+The pattern syntax is modified as follow:
+
+```antlr
+positional_pattern
+  : type? positional_pattern_clause length_pattern_clause? property_or_list_pattern_clause? simple_designation?
+  ;
```

Comment:
This is what's been discussed so far, I believe. cc @333fred

Reply:
Thanks. Confirmed with thread with LDM.